src/shared/KubectlDetector.cs

```diff
@@ -30,22 +30,36 @@ public static Task<bool> IsKubectlConnectedToClusterAsync(OutputContext output)
             return _kubectlConnectedToCluster.Value;
         }
 
-        private static async Task<bool> DetectKubectlInstalled()
+        private static async Task<Version?> GetKubectlVersion()
         {
             try
             {
                 // Ignoring the exit code and relying on Process to throw if kubectl is not found
                 // kubectl version will return non-zero if you're not connected to a cluster.
-                await ProcessUtil.RunAsync("kubectl", "version", throwOnError: false);
-                return true;
+                var result = await ProcessUtil.RunAsync("kubectl", "version", throwOnError: false);
+
+                var output = result.StandardOutput;
+                var serverStringIndex = output.IndexOf("Server Version: ");
+
+                var serverMajor = GetVersion(serverStringIndex, "Major:\"", "\"", output);
+                var serverMinor = GetVersion(serverStringIndex, "Minor:\"", "\"", output);
+                var version = new Version(serverMajor, serverMinor);
+                return version;
             }
             catch (Exception)
             {
                 // Unfortunately, process throws
-                return false;
+                return null;
             }
         }
 
+        private static int GetVersion(int startIndex, string start, string end, string input)
+        {
+            var first = input.IndexOf(start, startIndex) + start.Length;
```

Comment:
This is a hair hacky to parse a json object with string indexs, but I only needed two pieces of info from it.

Reply:
Wouldn't System.Text.Json work well here?

Reply:
Yeah, I realized I can output the kubectl version as json, making it easy to parse. Before, it wasn't valid json.